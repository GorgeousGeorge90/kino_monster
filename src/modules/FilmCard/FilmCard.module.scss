


.film_card_container {
  @apply flex flex-col gap-4 items-center mt-8;

  .rating_error {
    @apply text-red-600 text-sm font-bold font-sans-serif animate-open;
  }

  .film_card_text {
    @apply text-lg text-neutral-300;
  }
}