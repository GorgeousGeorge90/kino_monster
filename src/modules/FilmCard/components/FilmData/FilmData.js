import styles from './FilmData.module.scss';




const FilmData = ({Title,Year,Poster, Metascore,imdbRating,
    Released,Genre,Director,Actors,Plot,
}) => {

    return (<div className={styles.film_data_container}>
        <div className={styles.film_data_content}>
            <header className={styles.film_data_header}>
                <h3 className={styles.film_data_title}>{Title}</h3>
                <p className={styles.film_data_year}>{Year}</p>
            </header>
            <aside className={styles.film_data_left}>
                <div className={styles.film_data_poster}>
                    <img src={Poster} alt="poster"/>
                </div>
                <ul className={styles.film_data_scores}>
                    <li><b>metascore:</b> {Metascore}</li>
                    <li><b>imdb:</b> {imdbRating}</li>
                </ul>
            </aside>
            <main className={styles.film_data_main}>
                <section className={styles.film_data_plot}>
                    <article>
                        <p><b>Plot:</b> {Plot}</p>
                    </article>
                </section>
                <section>
                    <ul>
                        <li><p><b>released:</b> {Released}</p></li>
                        <li><p><b>genre:</b> {Genre}</p></li>
                        <li><p><b>director:</b> {Director}</p></li>
                        <li><p><b>actors:</b> {Actors}</p></li>
                    </ul>
                </section>
            </main>
        </div>
    </div>)
}

export default FilmData