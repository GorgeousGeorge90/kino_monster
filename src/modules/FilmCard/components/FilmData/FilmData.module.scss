


.container {
  @apply flex items-center justify-center font-sans-serif;

  .film_data_content {
    @apply grid grid-cols-[repeat(2,200px)] grid-rows-[50px,400px] justify-center gap-2;

    .film_data_header {
      @apply flex flex-col items-center col-span-full;

      .film_data_title {
        @apply font-bold text-neutral-100 text-lg;
      }

      .film_data_year {
        @apply  text-neutral-300 text-xs;
      }
    }

    .film_data_left {
      @apply flex flex-col items-start;

      .film_data_scores {
        @apply flex flex-col;
      }

      .film_data_poster {
        @apply self-start;

        img {
          @apply object-scale-down rounded-xl;
        }
      }

    }

    .film_data_main {
      @apply flex flex-col gap-3 font-sans-serif text-neutral-300 text-xs;

      .film_data_plot {
        @apply text-neutral-300 text-xs;
      }
    }
  }
}