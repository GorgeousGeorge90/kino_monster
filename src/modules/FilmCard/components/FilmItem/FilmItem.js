import styles from './FilmItem.module.scss';
import {useDispatch} from "react-redux";
import {deleteFilm, getFilm} from "../../store/searchSlice";
import {ReactComponent as TrashIcon} from '../../../../assets/img/trash1.svg';




const FilmItem = ({title,year,rating,id,poster}) => {
    const dispatch = useDispatch()

    return (<li className={styles.container}>
            <div className={styles.film_item_logo}>
                <img src={poster}
                     alt="logo"
                     onClick={()=>dispatch(getFilm(id))}
                />
            </div>
        <div className={styles.film_title_year}>
            <h3>{title}</h3>
            <p><span className={'font-bold'}>released:</span>{year}</p>
            <p>
                <span className={'font-bold'}>imdb:</span>
                <span>{rating}</span>
                <span className={styles.btn}
                      onClick={() => dispatch(deleteFilm(id))}>
                      <TrashIcon height='15px' width='15px'/>
                </span>
            </p>
        </div>

    </li>)
}

export default FilmItem