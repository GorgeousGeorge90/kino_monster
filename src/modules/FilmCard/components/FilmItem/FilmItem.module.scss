

.container {
  @apply flex items-center justify-center gap-1.5 border-none  rounded-md p-2 dark:bg-neutral-500
    hover:bg-amber-800;

  .film_item_logo {
    img {
      @apply h-20 rounded-md cursor-pointer;
    }

  }

  .film_title_year {
    @apply font-sans-serif relative;

    h3 {
      @apply font-bold text-xl;
    }

    p {

    }
  }

  .btn {
    @apply cursor-pointer absolute bottom-2 left-14;

  }
}