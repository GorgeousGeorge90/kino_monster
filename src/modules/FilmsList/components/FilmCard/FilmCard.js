import styles from './FilmCard.module.scss';
import CustomLink from "../../../../UI/CustomLink/CustomLink";
import {useDispatch} from "react-redux";
import {selectFilm} from "../../store/filmsSlice";
import pic from './img/null.png';


const FilmCard = ({id, title, poster_path, vote_average}) => {
    const dispatch = useDispatch()


    return (<li className={styles.film_card_container}>
        <CustomLink to={`/${id}`}>
            <img className={styles.film_card_poster}
                 src={`https://image.tmdb.org/t/p/original/${poster_path}`}
                 alt={`${pic}`}
                 onClick={()=>dispatch(selectFilm(id))}
            />
        </CustomLink>
        <p className={styles.film_card_rate}>{vote_average}</p>
    </li>)
}


export default FilmCard