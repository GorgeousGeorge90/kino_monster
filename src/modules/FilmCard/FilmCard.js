import {useSelector} from 'react-redux';
import {getSearch} from './selectors/selectors';
import styles from './FilmCard.module.scss';
import FilmSearchForm from "./components/FilmSearchForm/FilmSearchForm";
import {ReactComponent as Loader} from '../../assets/img/spinner.svg';
import FilmData from "./components/FilmData/FilmData";


const FilmCard = () => {
    const {film,status,error} = useSelector(getSearch)


    return (<section className={styles.container}>
        <h2>get new film</h2>
        <FilmSearchForm/>
        {
            error ? <span className={styles.rating_error}>{error}</span> : null
        }
        <h3>film info</h3>
        {
            !film ? <p>don't have any results</p>: status === 'pending' ? <Loader/>: <FilmData {...film}/>
        }
    </section>)
}

export default FilmCard