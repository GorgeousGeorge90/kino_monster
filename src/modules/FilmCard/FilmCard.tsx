import { getSearchError, getSearchFilm, getSearchStatus } from './selectors/selectors';
import styles from './FilmCard.module.scss';
import FilmSearchForm from "./components/FilmSearchForm/FilmSearchForm";
import { ReactComponent as Spinner } from '../../assets/img/spinner.svg';
import FilmItem from "./components/FilmItem/FilmItem";
import { useAppSelector } from '../../store/hooks';


const FilmCard = () => {
    const film = useAppSelector(getSearchFilm)
    const status = useAppSelector(getSearchStatus)
    const error = useAppSelector(getSearchError)

    return (<div className={styles.film_card_container}>
        <div className={styles.film_card_content}>
            <FilmSearchForm/>
            {
                error ? <span className={styles.rating_error}>{error}</span> : null
            }
            {
                !film ? <p className={styles.film_card_text}>don't have any requests</p> : status === 'pending' ?
                    <Spinner/> : <FilmItem {...film}/>
            }
        </div>
    </div>)
}

export default FilmCard