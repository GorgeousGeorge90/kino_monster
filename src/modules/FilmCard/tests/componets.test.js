import {render,screen} from '@testing-library/react';
import FilmData from "../components/FilmData/FilmData";
import * as reduxHooks from 'react-redux';
import FilmItem from "../components/FilmItem/FilmItem";
import FilmCard from "../FilmCard";

jest.mock('react-redux')
const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch')
const mockedSelector = jest.spyOn(reduxHooks, 'useSelector')

describe('render test FilmsSearch',()=> {
    it('goal:render FilmData component',()=> {

        const data = {
            Title:'title',
            Year:'year',
            Poster:'poster',
            imdbRating:'rating',
            Genre:'genre',
            Director:'director',
            Actors:'actors',
            Plot:'plot',
            Released:'released',
            Awards:'awards',
            Metascore:'metascore',
        }
        const dispatch = jest.fn()
        mockedDispatch.mockReturnValue(dispatch)
        mockedSelector.mockReturnValue(data)

        const view = render(<FilmData/>)
        expect(view).toMatchSnapshot()
    })

    it('goal:render FilmItem component',()=> {
        const data = {
            title:'title',
            year:'1990',
            rating:'8.7',
            id:'123',
            poster:'poster',
        }

        const dispatch = jest.fn()
        mockedDispatch.mockReturnValue(dispatch)
        render(<FilmItem {...data}/>)

        const film = screen.getByRole('listitem')
        expect(film).toBeInTheDocument()
        expect(film).toHaveTextContent('title')
    })

    // it('goal:render FilmsSearch component',()=>{
    //
    //     const data = {
    //         films: [
    //             {id:'1', title:'Hello!', year:'1990',rating:'9', poster:'pic'},
    //             {id:'2', title:'Hello2!', year:'1989',rating:'8', poster:'pic'},
    //         ],
    //         search: {
    //             currenFilm:null,
    //             error:null,
    //             status:'idle',
    //         }
    //     }
    //
    //     mockedSelector.mockReturnValue(data)
    //     render(<FilmsSearch/>)
    //
    //     const component = screen.getByRole('heading')
    //     expect(component).toBeInTheDocument()
    //     expect(component).toHaveTextContent('Top Films')
    //
    // })
})