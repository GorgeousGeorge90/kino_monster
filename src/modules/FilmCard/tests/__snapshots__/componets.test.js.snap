// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test FilmsSearch goal:render FilmData component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="film_data_container"
      >
        <div
          class="film_data_content"
        >
          <header
            class="film_data_header"
          >
            <h3
              class="film_data_title"
            />
            <p
              class="film_data_year"
            />
          </header>
          <aside
            class="film_data_left"
          >
            <div
              class="film_data_poster"
            >
              <img
                alt="poster"
              />
            </div>
            <ul
              class="film_data_scores"
            >
              <li>
                <b>
                  metascore:
                </b>
                 
              </li>
              <li>
                <b>
                  imdb:
                </b>
                 
              </li>
            </ul>
          </aside>
          <main
            class="film_data_main"
          >
            <section
              class="film_data_plot"
            >
              <article>
                <p>
                  <b>
                    Plot:
                  </b>
                   
                </p>
              </article>
            </section>
            <section>
              <ul>
                <li>
                  <p>
                    <b>
                      released:
                    </b>
                     
                  </p>
                </li>
                <li>
                  <p>
                    <b>
                      genre:
                    </b>
                     
                  </p>
                </li>
                <li>
                  <p>
                    <b>
                      director:
                    </b>
                     
                  </p>
                </li>
                <li>
                  <p>
                    <b>
                      actors:
                    </b>
                     
                  </p>
                </li>
              </ul>
            </section>
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="film_data_container"
    >
      <div
        class="film_data_content"
      >
        <header
          class="film_data_header"
        >
          <h3
            class="film_data_title"
          />
          <p
            class="film_data_year"
          />
        </header>
        <aside
          class="film_data_left"
        >
          <div
            class="film_data_poster"
          >
            <img
              alt="poster"
            />
          </div>
          <ul
            class="film_data_scores"
          >
            <li>
              <b>
                metascore:
              </b>
               
            </li>
            <li>
              <b>
                imdb:
              </b>
               
            </li>
          </ul>
        </aside>
        <main
          class="film_data_main"
        >
          <section
            class="film_data_plot"
          >
            <article>
              <p>
                <b>
                  Plot:
                </b>
                 
              </p>
            </article>
          </section>
          <section>
            <ul>
              <li>
                <p>
                  <b>
                    released:
                  </b>
                   
                </p>
              </li>
              <li>
                <p>
                  <b>
                    genre:
                  </b>
                   
                </p>
              </li>
              <li>
                <p>
                  <b>
                    director:
                  </b>
                   
                </p>
              </li>
              <li>
                <p>
                  <b>
                    actors:
                  </b>
                   
                </p>
              </li>
            </ul>
          </section>
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
